Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -1228,6 +1249,104 @@ bool ResourceLoadStatisticsStore::insertObservedDomain(const ResourceLoadStatist
+    };
+
+    auto scopedStatement = this->scopedStatement(m_insertLoginstatusStatement, insertLoginStatusQuery, "insertLoginstatus"_s);
+    if (!scopedStatement) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to create scoped statement");
+        return;
+    }
+
+    if (scopedStatement->bindText(RegistrableDomainIndex, domain.string()) != SQLITE_OK) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind registrableDomain");
+        return;
+    }
+
+    if (lastAuthentication) {
+        if (scopedStatement->bindText(UserNameIndex, lastAuthentication->username()) != SQLITE_OK
+            || scopedStatement->bindInt(TokenTypeIndex, static_cast<int>(lastAuthentication->tokenType())) != SQLITE_OK
+            || scopedStatement->bindInt(AuthenticationTypeIndex, static_cast<int>(lastAuthentication->authType())) != SQLITE_OK
+            || scopedStatement->bindDouble(LoggedInTimeIndex, lastAuthentication->loggedInTime().secondsSinceEpoch().value()) != SQLITE_OK
+            || scopedStatement->bindDouble(TimeToLiveIndex, lastAuthentication->timeToLive().seconds()) != SQLITE_OK) {
+            ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind lastAuthentication values");
+            return;
+        }
+    } else {
+        if (scopedStatement->bindNull(UserNameIndex) != SQLITE_OK
+            || scopedStatement->bindNull(TokenTypeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(AuthenticationTypeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(LoggedInTimeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(TimeToLiveIndex) != SQLITE_OK) {
+            ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind NULL values for lastAuthentication");
+            return;
+        }
+    }
+
+    if (scopedStatement->bindInt(StatusIndex, static_cast<int>(loggedInStatus)) != SQLITE_OK) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind status");
+        return;
+    }
+
+    if (scopedStatement->step() != SQLITE_DONE) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to step statement");
+        return;
+    }
+    return;
+}
+
+void ResourceLoadStatisticsStore::removeLoginStatus(RegistrableDomain&& domain)
+{
+    ASSERT(!RunLoop::isMain());
+
+    auto domainIDToRemove = loginStatusDomainID(WTFMove(domain));
+    if (!domainIDToRemove) {
+        ITP_RELEASE_LOG_ERROR("removeLoginStatus: failed to find domain");
+        return;
+    }
+
+    auto scopedStatement = this->scopedStatement(m_removeLoginStatusStatement, removeLoginStatusQuery, "removeLoginStatus"_s);
+    if (!scopedStatement
+        || scopedStatement->bindInt(1, *domainIDToRemove) != SQLITE_OK
+        || scopedStatement->step() != SQLITE_DONE)
+        ITP_RELEASE_LOG_DATABASE_ERROR("removeLoginstatus: failed to step statement");
+}
+
+bool ResourceLoadStatisticsStore::loginStatusExists(const RegistrableDomain&& domain)
```

Comment:
Shouldn't be `const` and an r-value reference. Should just use `const RegistrableDomain&` as mentioned above. This is causing Linux build breakage.